crates/bevy_ecs/src/query/iter.rs

```diff
@@ -69,6 +69,68 @@ where
             current_index: 0,
         }
     }
+
+    /// Consumes `self` and returns true if there were any elements remaining in this iterator.
+    #[inline(always)]
+    pub(crate) fn any_remaining(mut self) -> bool {
```

Comment:
Also, is this something I should just make `pub` for `QueryIter`?

Reply:
for now lets just leave it private. we can expose it if a need comes up, but this feels pretty niche to me.